python/hsfs/feature_store.py

```diff
@@ -903,6 +903,146 @@ def create_external_feature_group(
             online_enabled=online_enabled,
         )
 
+    def get_or_create_spine_group(
+        self,
+        name: str,
+        version: Optional[int] = None,
+        description: Optional[str] = "",
+        primary_key: Optional[List[str]] = [],
+        event_time: Optional[str] = None,
+        features: Optional[List[feature.Feature]] = [],
+        dataframe: Union[
+            pd.DataFrame,
+            TypeVar("pyspark.sql.DataFrame"),  # noqa: F821
+            TypeVar("pyspark.RDD"),  # noqa: F821
+            np.ndarray,
+            List[list],
+        ] = None,
+    ):
+        """Create a spine group metadata object.
+
+        Instead of using a feature group to save a label/prediction target, you can use a spine together with a dataframe containing the labels.
+        A Spine is essentially a metadata object similar to a feature group, however, the data is not materialized in the feature store.
+        It only containes the needed metadat such as the relevant event time column and primary key columns to perform point-in-time correct joins.
+
+        !!! example
+            ```python
+            # connect to the Feature Store
+            fs = ...
+
+            spine_df = pd.Dataframe()
+
+            spine_group = fs.get_or_create_spine_group(
+                                name="sales",
+                                version=1,
+                                description="Physical shop sales features",
+                                primary_key=['ss_store_sk'],
+                                event_time='sale_date',
+                                dataframe=spine_df
+                                )
+            ```
+
+        Note that you can inspect the dataframe in the spine group, or replace the dataframe:
+
+        ```python
+        spine_group.dataframe.show()
+
+        spine_group.dataframe = new_df
+        ```
+
+        The spine can then be used to construct queries, with only one speciality:
+
+        !!! note
+            Spines can only be used on the left side of a feature join, as this is the base
+            set of entities for which features are to be fetched and the left side of the join
+            determines the event timestamps to compare against.
+
+        **If you want to use the query for a feature view to be used for online serving,
+        you can only select the label or target feature from the spine.**
```

Comment:
I guess you mean that if you are using the spine, then the above applies. But if you are not using the spine the label can come from any feature group.

Reply:
Spines can always only be on the left side of a join. If you want to do online serving -> you can only select the label.
Of course if you don't use a spine at all, you can use any column as label. but this docs deals only with spines, so I am not sure what you are asking to change?